src/commands/external.rs

```diff
@@ -123,6 +88,98 @@ pub async fn execute_external_subcommand(
     Ok(())
 }
 
+async fn ensure_plugin_available(
+    plugin_name: &str,
+    plugin_store: &PluginStore,
+    app: clap::App<'_>,
+    override_compatibility_check: bool,
+) -> anyhow::Result<Option<String>> {
+    let plugin_version = match plugin_store.read_plugin_manifest(plugin_name) {
+        Ok(manifest) => {
+            if let Err(e) =
+                warn_unsupported_version(&manifest, SPIN_VERSION, override_compatibility_check)
+            {
+                eprintln!("{e}");
+                // TODO: consider running the update checked?
+                process::exit(1);
+            }
+            Some(manifest.version().to_owned())
+        }
+        Err(PluginError::NotFound(e)) => {
+            consider_install(plugin_name, plugin_store, app, &e).await?
+        }
+        Err(e) => return Err(e.into()),
+    };
+    Ok(plugin_version)
+}
+
+async fn consider_install(
+    plugin_name: &str,
+    plugin_store: &PluginStore,
+    app: clap::App<'_>,
+    e: &spin_plugins::error::NotFoundError,
+) -> anyhow::Result<Option<String>> {
+    if predefined_externals()
```

Comment:
A thought for a potential follow-up as I acquaint myself with this function.  I see it currently tracks `cloud`: https://github.com/fermyon/spin/blob/362c75d98e3ac2cb1b760b661fbcb18f0eced118/src/commands/external.rs#L39-L42

Should we add js2wasm and py2wasm per https://github.com/fermyon/developer/blob/main/downloads/install.sh#L148?

Reply:
We predefine `cloud` because it is the backing implementation for two hardwired commands, `spin deploy` and `spin login`.  This isn't the case for `js2wasm` and `py2wasm`.